### PR TITLE
modernize scanmach.d

### PR DESCRIPTION
**Pull request by WalterBright**
Not much here.

### CODE REVIEW

File: compiler/src/dmd/scanmach.d [hunk 1]
```diff
@@ -33,7 +33,7 @@ private enum LOG = false;
  *      eSink =       where the error messages go
  */
 void scanMachObjModule(void delegate(const(char)[] name, int pickAny) pAddSymbol,
-        const(ubyte)[] base, const(char)* module_name, Loc loc, ErrorSink eSink)
+        scope const ubyte[] base, scope const char* module_name, Loc loc, ErrorSink eSink)
```
Comment (dkorpel) [inline compiler/src/dmd/scanmach.d:36]: I'm wary of adding `scope` to parameters without the compiler verifying it. I think `eSink` can escape `module_name`, which would make the annotation incorrect.

Comment (WalterBright) [reply compiler/src/dmd/scanmach.d:36]: You're right that the compiler cannot check use of ... arguments for safety. The eSink functions would have to be marked @trusted. Also, the ... parameter should be prefixed with `scope`. The current compiler does not check this properly, so I added:

https://issues.dlang.org/show_bug.cgi?id=24105

But, in anticipation of fixing 24105, I'd like to leave the intent in this code in. BTW, due to bootstrapping, dmd isn't going to have up to date checking anyway.

Comment (dkorpel) [reply compiler/src/dmd/scanmach.d:36]: In the absence of compiler checks, at least the `scope` annotations should still be correct when checked manually, so eSink's `...` parameters must be marked `scope`. A `scope` parameter should mean I can rely on the function to not escape the parameter, not that the code intents "Can the parameter maybe not escape the function please? That would be great."

Wrong `scope` annotations can cause (and have caused) memory corruption.

> BTW, due to bootstrapping, dmd isn't going to have up to date checking anyway.

The CI only passes when *both* the bootstrap and the new compiler can compile dmd. Even when the older one doesn't have the updated checks, the newer one will still catch `scope` violations.

Comment (WalterBright) [reply compiler/src/dmd/scanmach.d:36]: BTW, `scope` has already been added to all the `...` parameters for the `printf` family.

Comment (WalterBright) [reply compiler/src/dmd/scanmach.d:36]: https://github.com/dlang/dmd/pull/15556 should address your concern.

Comment (WalterBright) [reply compiler/src/dmd/scanmach.d]: removed the scope

**Review by dkorpel — APPROVED** [review]

**Pull request merged**